org.lflang/src/org/lflang/generator/CppGenerator.kt

```diff
@@ -0,0 +1,263 @@
+                |   GIT_TAG "$runtimeVersion"
+                |   CMAKE_ARGS
+                |   -DCMAKE_BUILD_TYPE:STRING=$S{CMAKE_BUILD_TYPE}
+                |   -DCMAKE_INSTALL_PREFIX:PATH=$S{CMAKE_INSTALL_PREFIX}
+                |   -DCMAKE_INSTALL_BINDIR:PATH=$S{CMAKE_INSTALL_BINDIR}
+                |   -DCMAKE_CXX_COMPILER=$S{CMAKE_CXX_COMPILER}
+                |   -DREACTOR_CPP_VALIDATE=${if (targetConfig.noRuntimeValidation) "OFF" else "ON"}
+                |   -DREACTOR_CPP_TRACE=${if (targetConfig.tracing != null) "ON" else "OFF"} 
+                |   -DREACTOR_CPP_LOG_LEVEL=${targetConfig.logLevel.severity}
+                |)
+                |
+                |set(REACTOR_CPP_LIB_DIR "$S{CMAKE_INSTALL_PREFIX}/$S{CMAKE_INSTALL_LIBDIR}")
+                |set(REACTOR_CPP_BIN_DIR "$S{CMAKE_INSTALL_PREFIX}/$S{CMAKE_INSTALL_BINDIR}")
+                |set(REACTOR_CPP_LIB_NAME "$S{CMAKE_SHARED_LIBRARY_PREFIX}reactor-cpp$S{CMAKE_SHARED_LIBRARY_SUFFIX}")
+                |set(REACTOR_CPP_IMPLIB_NAME "$S{CMAKE_STATIC_LIBRARY_PREFIX}reactor-cpp$S{CMAKE_STATIC_LIBRARY_SUFFIX}")
+                |
+                |add_library(reactor-cpp SHARED IMPORTED)
+                |add_dependencies(reactor-cpp dep-reactor-cpp)
+                |if(WIN32)
+                |   set_target_properties(reactor-cpp PROPERTIES IMPORTED_IMPLIB "$S{REACTOR_CPP_LIB_DIR}/$S{REACTOR_CPP_IMPLIB_NAME}")
+                |   set_target_properties(reactor-cpp PROPERTIES IMPORTED_LOCATION "$S{REACTOR_CPP_BIN_DIR}/$S{REACTOR_CPP_LIB_NAME}")
+                |else()
+                |   set_target_properties(reactor-cpp PROPERTIES IMPORTED_LOCATION "$S{REACTOR_CPP_LIB_DIR}/$S{REACTOR_CPP_LIB_NAME}")
+                |endif()
+                |
+                |if (APPLE)
+                |   file(RELATIVE_PATH REL_LIB_PATH "$S{REACTOR_CPP_BIN_DIR}" "$S{REACTOR_CPP_LIB_DIR}")
+                |   set(CMAKE_INSTALL_RPATH "@executable_path/$S{REL_LIB_PATH}")
+                |else ()
+                |   set(CMAKE_INSTALL_RPATH "$S{REACTOR_CPP_LIB_DIR}")
+                |endif ()
+            """.trimIndent()
+        }
+            |
+            |set(CMAKE_BUILD_WITH_INSTALL_RPATH ON)
+            |
+            |set(LF_MAIN_TARGET ${topLevelName})
+            |
+            |add_executable($S{LF_MAIN_TARGET}
+            |    main.cc
+        ${" |    "..sourceFiles.joinToString("\n")}
+            |)
+            |target_include_directories($S{LF_MAIN_TARGET} PUBLIC
+            |    "$S{CMAKE_INSTALL_PREFIX}/$S{CMAKE_INSTALL_INCLUDEDIR}"
+            |    "$S{PROJECT_SOURCE_DIR}"
+            |    "$S{PROJECT_SOURCE_DIR}/__include__"
+            |)
+            |target_link_libraries($S{LF_MAIN_TARGET} reactor-cpp)
+            |
+            |install(TARGETS $S{LF_MAIN_TARGET})
+        ${
+            if (includeFile == null) "" else """
+                |
+                |include($includeFile)""".trimIndent()
```

Comment:
same for this `trimIndent`.
Also this dollar block could be written `${ includeFile?.let { "|include($it)" }.orEmpty() }`